src/Datadog.Trace/DiagnosticListeners/IEndpointFeature.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.DiagnosticListeners
+{
+    /// <summary>
+    /// IEndpointFeature for ducktyping HttpContext
+    /// </summary>
+    public interface IEndpointFeature
+    {
+        /// <summary>
+        /// Delegates to IEndpointFeature.Endpoint;
+        /// </summary>
+        [Duck(Name = "Microsoft.AspNetCore.Http.Features.IEndpointFeature.get_Endpoint")]
+        RouteEndpoint GetEndpoint();
```

Comment:
Is there a reason you're proxying the underlying method instead of proxying the property itself?

Reply:
Yes! 🙂 

The default feature collection implementation used by Kestrel [explicitly implements the interface here](https://github.com/dotnet/aspnetcore/blob/28a6f89eb5ca73c631bda39f8b8d690cc2c7eeb4/src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs#L21) . Unfortunately ducktyping can't find this method, even with updated binding flags, the only way I could see to make it work was to get the method, as in this case.

Note that this is implementation specific; the `TestServer` implementation for example _can_ proxy the property directly, hence the additional `EndpointFeatureStruct` definition: https://github.com/DataDog/dd-trace-dotnet/blob/ba9e30544e280200fc61091f3f577819950d0c45/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs#L882-L886